interbotix_ros_xscobots/interbotix_xscobot_moveit/launch/moveit_api.launch.py

```diff
@@ -0,0 +1,333 @@
+    get_interbotix_xscobot_models,
+)
+
+def load_yaml(package_name, file_path):
+    package_path = get_package_share_directory(package_name)
+    absolute_file_path = os.path.join(package_path, file_path)
+
+    try:
+        with open(absolute_file_path, 'r') as file:
+            return yaml.safe_load(file)
+    except EnvironmentError:  # parent of IOError, OSError *and* WindowsError where available
+            return None
+
+def launch_setup(context, *args, **kwargs):
+
+    robot_model_launch_arg = LaunchConfiguration('robot_model')
+    robot_name_launch_arg = LaunchConfiguration('robot_name')
+    base_link_frame_launch_arg = LaunchConfiguration('base_link_frame')
+    use_world_frame_launch_arg = LaunchConfiguration('use_world_frame')
+    external_urdf_loc_launch_arg = LaunchConfiguration('external_urdf_loc')
+    mode_configs_launch_arg = LaunchConfiguration('mode_configs')
+    use_moveit_rviz_launch_arg = LaunchConfiguration('use_moveit_rviz')
+    rviz_frame_launch_arg = LaunchConfiguration('rviz_frame')
+    rviz_config_file_launch_arg = LaunchConfiguration('rviz_config_file')
+    world_filepath_launch_arg = LaunchConfiguration('world_filepath')
+    robot_description_launch_arg = LaunchConfiguration('robot_description')
+    hardware_type_launch_arg = LaunchConfiguration('hardware_type')
+    xs_driver_logging_level_launch_arg = LaunchConfiguration('xs_driver_logging_level')
+
+    use_sim_time_param = determine_use_sim_time_param(
+        context=context,
+        hardware_type_launch_arg=hardware_type_launch_arg
+    )
+
+    remappings = [
+        (
+            f'{robot_name_launch_arg.perform(context)}/get_planning_scene',
+            f'/{robot_name_launch_arg.perform(context)}/get_planning_scene'
+        ),
+        (
+            '/arm_controller/follow_joint_trajectory',
+            f'/{robot_name_launch_arg.perform(context)}/arm_controller/follow_joint_trajectory'
+        ),
+        (
+            '/gripper_controller/follow_joint_trajectory',
+            f'/{robot_name_launch_arg.perform(context)}/gripper_controller/follow_joint_trajectory'
+        ),
+        ]
+
+    planning_scene_monitor_parameters = {
+        'publish_planning_scene': True,
+        'publish_geometry_updates': True,
+        'publish_state_updates': True,
+        'publish_transforms_updates': True,
+    }
+
+    moveit_config = (
+        MoveItConfigsBuilder(
+            robot_name="dx400", package_name="interbotix_xscobot_moveit"
+        )
+        .robot_description(file_path="config/dx400.urdf.xacro")
+        .robot_description_semantic(file_path="config/srdf/dx400.srdf.xacro")
+        .trajectory_execution(file_path="config/moveit_controllers.yaml")
+        .joint_limits(file_path="config/joint_limits/dx400_joint_limits.yaml")
+        .robot_description_kinematics(file_path="config/moveit_kinematics.yaml")
+        .planning_scene_monitor(planning_scene_monitor_parameters)
+        # .planning_pipelines(pipelines=["ompl"])
```

Comment:
Why is this line commented? Was this the issue we were seeing before where it wasn't playing nicely with the planning pipelines?

Reply:
Yes, So I wanted to try out different pipelines and use another pipeline than `ompl` but it somehow never worked so we ended up working with `ompl` which is specified in the  yaml file.